src/site/content/en/fast/reduce-network-payloads-using-text-compression/index.md

```diff
@@ -156,3 +158,35 @@ const app = express();
 
 app.use(express.static('public'));
 </pre>
+
+## Effects on Core Web Vitals
+
+Depending the type of resource, both compression and minification can have a variety of effects on Core Web Vitals. Since both minification and compression reduce the payload size of the delivered resource, it's important to understand what Core Web Vitals are affected by these performance optimizations.
+
+### CSS
+
+When CSS resources are compressed and minified, a number of performance improvements may be observed in your Core Web Vitals metrics.
+
+- Where [First Contentful Paint (FCP)](/fcp/) is concerned, making your CSS as small as it can possibly be means that the browser will take less time to download the resource. This means that parsing of CSS can occur sooner, which should move up the rendering step.
+- Where [Largest Contentful Paint (LCP)](/lcp/) is concerned, it largely depends. If your LCP resource is specified in a CSS `background-image` property, making your CSS smaller means that the potential LCP resource can be discovered more quickly. However, LCP candidates that come from a CSS `background-image` property are not always optimal for LCP, as external CSS resources [don't benefit from the browser preload scanner](/preload-scanner/#css-background-images). If you can't move an LCP image out of CSS and into your HTML, then you should be preloading that image resource.
+
+### JavaScript
+
+JavaScript is a bit tricky, as it depends on what your JavaScript resources are providing for your website.
+
+- If your JavaScript is providing interactivity for your website, minification and compression will ensure that your website's [First Input Delay (FID)](/fid/) and [Interaction to Next Paint (INP)](/inp/) metrics may be improved during startup. However, it's also important that you're only serving the JavaScript that a given page needs, otherwise you may risk delays in responding to user inputs during that crucial startup period.
```

Comment:
Is this true? Until the JavaScript is run, the event handlers aren't attached so they have no impact on FID and INP.

I'm not sure compression and minification causes that big a difference to FID and INP to be honest. It will be uncompressed after downloading. And as it's parsed and compiled it's basically minified anyway. Yeah parsing minified code is likely a bit more efficient but not sure it's noticeable.

And on that note, doesn't JS always provide interactivity? And if it doesn't you don't care about FID and INP so think first clause is a bit redundant?

What about this:

```suggestion
Compressing and minifying JavaScript can have positive impacts on your website:

- Making the JavaScript resources smaller allows the JavaScript to download faster, allowing it to be to be processed earlier, allowing your page to become interactive earlier. However, it's also important that you're only serving the JavaScript that a given page needs, otherwise you may risk delays in responding to user inputs during that crucial startup period.
```

On this one I don't think there's an actual FID/INP benefit so not mentioning the metrics specifically.